Build error when using `DLMS_IGNORE_SERVER`

## Desription

When using `DLMS_IGNORE_SERVER` in `ArduinoIgnore.h` to disable DLMS server, the build fails with an error (unterminated brace `}`)

## Error Log

```text

[Starting] Verifying sketch 'gurux_arduino.ino'
Please see the build logs in output path: ~/XLabs/Projects/gurux_arduino/build

~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:1012:2: error: expected '}' at end of input
 1012 | }
      |  ^
In file included from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:37,
                 from ~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:40,
                 from ~/XLabs/Projects/gurux_arduino/gurux_arduino.ino:36:
~/Arduino/libraries/gx_dlms_c/src/include/gxinvoke.h:37:12: note: to match this '{'
   37 | extern "C" {
      |            ^
~/XLabs/Projects/gurux_arduino/GXDLMSClient.cpp:377:2: error: expected '}' at end of input
  377 | }
      |  ^
In file included from ~/Arduino/libraries/gx_dlms_c/src/gx_dlms_c.h:37,
                 from ~/XLabs/Projects/gurux_arduino/GXDLMSClient.h:40,
                 from ~/XLabs/Projects/gurux_arduino/GXDLMSClient.cpp:35:
~/Arduino/libraries/gx_dlms_c/src/include/gxinvoke.h:37:12: note: to match this '{'
   37 | extern "C" {
      |            ^

Error during build: exit status 1
IntelliSense configuration already up to date. To manually rebuild your IntelliSense configuration run "Ctrl+Alt+I"
[Error] Verifying sketch 'gurux_arduino.ino': Exit with code=1

```

### RCA

- Building with all swithes enabled doesn't result in an error, so it's clear that this error is specific to the `DLMS_IGNORE_SERVER` switch.
- Checking `server.h` and `serverevents.h` yielded no results, both files are syntactically correct.
- After fidding around with `gx_dlms_c.h` (removing linkage scope), and compiling, the error appeared in `gxinvoke.h` due to an overlap between the header guard and the linkage scope definitions.

### Proposed solution

- Fix overlap between the header guard and the `DLMS_IGNORE_SERVER` switch in `gxinvoke.h`

--- a/src/include/gxinvoke.h
+++ b/src/include/gxinvoke.h
@@ -62,9 +62,9 @@ extern "C" {
         gxValueEventArg* e);
 #endif //DLMS_IGNORE_SCRIPT_TABLE
 
+#endif //DLMS_IGNORE_SERVER
 #ifdef  __cplusplus
 }
 #endif
-#endif //DLMS_IGNORE_SERVER
 
 #endif //COSEM_INVOKE_H

